src/model/scripts/llama_thai_tokenizer/README.md

```diff
@@ -1,25 +1,47 @@
-# Tokenizer Experiment
+# OpenThaiGPT Merged Tokenizer Pipeline
 
-merge LLaMa Tokenizer with [wangchanberta](https://huggingface.co/airesearch/wangchanberta-base-att-spm-uncased/tree/main) Tokenizer by extent vocabulary following [working example](https://github.com/ymcui/Chinese-LLaMA-Alpaca/blob/main/README_EN.md) 
+we plan to use [Pretrained LLaMA Model](https://ai.meta.com/blog/large-language-model-llama-meta-ai/) as a base model for finetuning but LLaMa Tokenizer (BPE Tokenizer) have less Thai vocabulary, this pipeline intent to extent vocabulary of LlaMA tokenizer, using merge method following [Chinese-LLaMA-Alpaca](https://github.com/ymcui/Chinese-LLaMA-Alpaca/blob/main/README_EN.md)
 
-## Objective
-1. have faster iteration when tokenize Thai and English sentence.
-2. still have similar tokenized output when running with English sentence.
+## Method
 
-## To save merge tokenizer
+- merge LLaMA tokenizer by extenting vocabulary and merge rule from BPE Thai tokenizer
+  ![method](merge_method.png)
 
-```python3 scripts/llama_thai_tokenizer/merge_tokenizer.py```
+## merge and save merge tokenizer
+
+1. load LLaMA tokenizer by pass your model name to an argument
+
+```bash
+python3 src/model/scripts/llama_thai_tokenizer/load_tokenizer.py --model_name meta-llama/Llama-2-7b --output_path path/to/save
+```
+
+argument
+
+- model_name: name of LLaMA model that you use
+- output_path: path to save
+
+2. merge tokenizer `need to have Thai BPE tokenizer first` then add path of it be an argument of following script
+
+```bash
+python3 scripts/llama_thai_tokenizer/merge_tokenizer.py --llama_path path/to/model --thai_sp_path path/to/thai_model --output_path path/to/save
+```
+
+argument
+
+- llama_path: path to LLaMA tokenizer
```

Comment:
When you mentioned path, does this mean `model.spm` or `huggingface_tokenizer_path` (folder)?

Can you elaborate on it here in this argument for each paths?

Reply:
done